docs/topics/yaml-rule-structure-syntax.adoc

```diff
@@ -0,0 +1,520 @@
++
+`--label-selector="key1=val1 && key2"`
+
+* To perform logical OR operations on matches of multiple rules using the `||` operator:
++
+`--label-selector="key1=val1 || key2"`
+
+* To perform a NOT operation to filter-out rules that have `key1=val1` label set using the `!` operator:
++
+`--label-selector="!key1=val1"`
+
+* To group sub-expressions and control precedence using AND:
++
+`--label-selector="(key1=val1 || key2=val2) && !val3"`
+
+.Dependency labels
+
+The analyzer engine adds labels to dependencies. These labels provide additional information about a dependency, such as its programming language and whether the dependency is open-source or internal.
+
+Currently, the analyzer adds the following labels to dependencies:
+
+[source,terminal]
+----
+labels:
+- konveyor.io/dep-source=internal
+- konveyor.io/language=java
+----
+
+.Dependency label selector
+
+The analyzer CLI accepts the `--dep-label-selector` option, which allows filtering-in or filtering-out incidents generated from a dependency by their labels.
+
+For example, the analyzer adds a `konveyor.io/dep-source` label to dependencies with a value that indicates whether the dependency is a known open-source dependency.
+
+To exclude incidents for all such open-source dependencies, you can use `--dep-label-selector` as follows:
+
+`konveyor-analyzer ... --dep-label-selector !konveyor.io/dep-source=open-source`
+
+The Java provider in the analyzer can also add an exclude label to a list of packages. To exclude all such packages, you can use `--dep-label-selector` as follows:
+
+`konveyor-analyzer ... --dep-label-selector !konveyor.io/exclude`
+
+[id="yaml-rule-categories_{context}"]
+=== Rule categories
+
+* `mandatory`
+** You must resolve the issue for a successful migration, otherwise, the resulting application is not expected to build or run successfully. An example of such an issue is proprietary APIs that are not supported on the target platform.
+* `optional`
+** If you do not resolve the issue, the application is expected to work, but the results might not be optimal. If you do not make the change at the time of migration, you need to put it on the schedule soon after your migration is completed. An example of such an issue is EJB 2.x code not upgraded to EJB 3.
+* `potential`
+** You need to examine the issue during the migration process, but there is not enough information to determine whether resolving the issue is mandatory for the migration to succeed. An example of such an issue is migrating a third-party proprietary type when there is no directly compatible type on the target platform.
+
+
+[id="yaml-rule-actions_{context}"]
+=== Rule Actions
+
+Rules can include 2 types of actions: _message_ and _tag_. Each rule includes one of them or both.
+
+.Message actions
+
+The message action creates an issue with a message when the rule matches. The custom data exported by providers can also be used in the message.
```

Comment:
"A message action..." First mention of a single item usually takes an indefinite article; the second and later references often take the definite article.